### ISSUE TITLE
[QUESTION] Why the require_grad attribute of the teacher model is True

### ISSUE THREAD

**Issue by MSXYljw** [lightly-ai/lightly-train#352]
I printed the require_grad attribute of the teacher model and I found all of them was set to true. I want to know why this is set up this way.

I was using lightly-train to train my local yolo11n model from a dinov3/vits16 model, and this is my training code

```
import lightly_train
from ultralytics import YOLO
model = YOLO("yolo11n.yaml")
if __name__ == "__main__":
    lightly_train.train(
        out="out/my_experiment", 
        overwrite=True,
        data="/root/autodl-tmp/data/pcb_yolo_format/images",
        model=model,
        method="distillationv1",
        batch_size=8,
        method_args={
            "teacher": "dinov3/vits16",
            "teacher_url": "a url",
        }
    )
```

then, I modified part of your source code to print the requires_grad attribute. at the beginning of `training_step_impl ` function in `Distillation` class, I wrote:

```      
    for name, para in self.named_parameters():
            print(name, para.requires_grad)
```

and it prints:
teacher_embedding_model.mask_token True
teacher_embedding_model.patch_embed.proj.weight True
teacher_embedding_model.patch_embed.proj.bias True
teacher_embedding_model.blocks.0.norm1.weight True
teacher_embedding_model.blocks.0.norm1.bias True
………

I think the require_grad attribute of the teacher model should be set to False. Would you mind explaining the reasoning behind this setup?

**Comment by stegmuel** [comment]
Hey,

Thanks for pointing this out.
The `requires_grad` attribute of teacher's parameters should indeed be `False`.
In practice the method behaves as intended (without memory/time overhead) as:

1. We use the `torch.no_grad` decorator for the [teacher forward pass](https://github.com/lightly-ai/lightly-train/blob/f32cd12cd03c9a12d630ef1dd7c8cec4235bf10b/src/lightly_train/_methods/distillation/distillation.py#L240)
2. The teacher parameters are [not passed to the optimizer
](https://github.com/lightly-ai/lightly-train/blob/f32cd12cd03c9a12d630ef1dd7c8cec4235bf10b/src/lightly_train/_methods/distillation/distillation.py#L289)
3. The teacher is not in training mode

In the future, we might explicitly set them to False to avoid confusion, but the current behavior is functionally identical.

**Comment by MSXYljw** [comment]
Hi,@stegmuel .Thank you for your response, it truly clarified my questions.
